docs/pnp/migrate-vsto-to-office-web-add-in-shared-code-library-tutorial.md

```diff
@@ -0,0 +1,427 @@
+
+### Document code
+
+In VSTO code interacts with the document through .NET objects such as `Microsoft.Office.Interop.Excel.Range`. But Office web add-ins use the Office.js library. Although these are similar, they are not exactly the same. So again, you cannot share document interaction code to the Office web add-in.
+
+### Logic code
+
+Business logic, algorithms, helper functions, and similar code often make up the heart of a VSTO Add-in. This code works independently of the UI and document code to perform analysis, connect to backend services, run calculations, and more. This is the code that can be shared so that you don't have to rewrite it in JavaScript.
+
+Let's examine the VSTO Add-in. In the following code, each section is identified as DOCUMENT, UI, or ALGORITHM code.
+
+```csharp
+// *** UI CODE ***
+private void btnUnicode_Click(object sender, EventArgs e)
+{
+    // *** DOCUMENT CODE ***
+    Microsoft.Office.Interop.Excel.Range rangeCell;
+    rangeCell = Globals.ThisAddIn.Application.ActiveCell;
+
+    string cellValue = "";
+
+    if (null != rangeCell.Value)
+    {
+        cellValue = rangeCell.Value.ToString();
+    }
+
+    // *** ALGORITHM CODE ***
+    //convert string to Unicode listing
+    string result = "";
+    foreach (char c in cellValue)
+    {
+        int unicode = c;
+
+        result += $"{c}: {unicode}\r\n";
+    }
+    
+    // *** UI CODE ***
+    //Output the result
+    txtResult.Text = result;
+}
+```
+
+Using this approach you can see that one section of code can be shared to the Office web add-in. The following code will need to be refactored into a separate class library.
+
+```csharp
+// *** ALGORITHM CODE ***
+//convert string to Unicode listing
+string result = "";
+foreach (char c in cellValue)
+{
+    int unicode = c;
+
+    result += $"{c}: {unicode}\r\n";
+}
+```
+
+## Create a shared class library
+
+VSTO Add-ins are created in Visual Studio as .NET projects, so we'll reuse .NET as much as possible to keep things simple. Our next technique is to create a class library and refactor shared code into that class library.
+
+1. If you haven't already, start Visual Studio 2019 and open the **\start\Cell-Analyzer.sln** solution.
+2. Right-click the solution in **Solution Explorer** and choose **Add > New Project**.
+3. In the **Add a new project dialog**, choose **Class Library (.NET Framework)** and choose **Next**.
+    > [!NOTE]
+    > Don't use the .NET Core class library because it will not work with your VSTO project.
+5. In the **Configure your new project** dialog, set the following fields.
+    - Set the  **Project name** to **CellAnalyzerSharedLibrary**.
+    - Leave the **Location** at it's default value.
+    - Set the **Framework** to **4.7.2**.
+6. Choose **Create**.
+7. After the project is created, rename the **Class1.cs** file to **CellOperations.cs**. You will be prompted to rename the class as well. Do that so the class name matches the file name.
```

Comment:
```suggestion
7. After the project is created, rename the **Class1.cs** file to **CellOperations.cs**. A prompt to rename the class appears. Rename the class name so that it matches the file name.
```